sync/class.jetpack-sync-actions.php

```diff
@@ -188,20 +183,38 @@ static function send_data( $data, $codec_name, $sent_timestamp, $queue_id, $chec
 		return $response;
 	}
 
-	static function do_initial_sync( $new_version = null, $old_version = null ) {
+	static function do_initial_sync_on_version_update( $new_version = null, $old_version = null ) {
 		if ( ! empty( $old_version ) && version_compare( $old_version, '4.2', '>=' ) ) {
 			return;
 		}
 
+		// Lets not sync if we are not suppose to.
+		if ( ! self::sync_allowed() ) {
+			return false;
+		}
+		self::do_initial_sync();
+	}
+
+	static function do_initial_sync() {
+		self::initialize_listener();
+		self::initialize_sender();
+		add_action( 'shutdown', array( self::$sender, 'do_full_sync' ) );
+
 		$initial_sync_config = array(
 			'options'         => true,
-			'network_options' => true,
 			'functions'       => true,
 			'constants'       => true,
-			'users'           => 'initial',
 		);
 
-		self::do_full_sync( $initial_sync_config );
+		if ( is_multisite() ) {
+			$initial_sync_config['network_options'] = true;
+		}
+
+		$user = wp_get_current_user();
+		if ( isset( $user->ID ) ) {
+			$initial_sync_config['users'] = array( $user->ID );
+		}
+		Jetpack_Sync_Modules::get_module( 'full-sync' )->start( $initial_sync_config );
```

Comment:
we could do `self::do_full_sync( $initial_sync_config );` here

Reply:
Nope since at this point we don't have all the info to request for sync to work. So doing the force method work as expected for now.

Reply:
I think we should try to use `do_full_sync` so we don't have many different points on the code were we can initiate a "full sync"

I would fix `sync_allowed` to return `true` if `doing_action( 'jetpack_user_authorized' )`